src/apps/backend/modules/todo/internal/todo_writer.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import asdict, replace
+from datetime import datetime
+
+from bson import ObjectId
+from pymongo import ReturnDocument
+
+from modules.todo.errors import InvalidDueDateFormatError, TodoNotFoundError
+from modules.todo.internal.store.todo_model import TodoModel
+from modules.todo.internal.store.todo_repository import TodoRepository
+from modules.todo.types import CreateTodoParams, Todo, UpdateTodoParams
+
+
+class TodoWriter:
+    @staticmethod
+    def create_todo(account_id: str, params: CreateTodoParams) -> Todo:
+        if isinstance(params.due_date, str):
+            try:
+                parsed_due_date = datetime.fromisoformat(params.due_date)
+            except ValueError:
+                raise InvalidDueDateFormatError(params.due_date)
+            params = replace(params, due_date=parsed_due_date)
+
+        todo_bson = TodoModel(
+            id=None,
```

Comment:
Can you please check and learn from the architecture [here](https://github.com/jalantechnologies/flask-react-template/blob/8d6d868806019d17922bf954487d6b78fa6d79d6/src/apps/backend/modules/account/internal/account_writer.py#L51)?